### PR TITLE
e2e regression test fixes 3 17

### PR DESCRIPTION
**Pull request by uwedeportivo**
collecting all the changes i'm making to get regression tests in shape (i'm a bumbling ts idiot, so...)

### CODE REVIEW

**Comment by uwedeportivo** [comment]
so far: extensions test for codecov is not finding the "Coverage 80%", no matter what i try. helping it out by actually clicking it myself (as a human) makes it pass. verified functionality by hand with @slimsag as a witness.

**Comment by uwedeportivo** [comment]
search tests have the usual failures where the search url composition is in random order that is hard write a expected value for

**Comment by uwedeportivo** [comment]
config-setttings doesn't work. complains that a second builtin auth provider is forbidden

File: web/src/regression/core.test.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ describe('Core functionality regression test suite', () => {
             await driver.page.waitForSelector('.e2e-settings-file .monaco-editor')
             return driver.page.evaluate(() => {
                 const editor = document.querySelector('.e2e-settings-file .monaco-editor') as HTMLElement
-                return editor ? editor.textContent : null
+                return editor ? editor.innerText : null
```
Comment (felixfbecker) [inline web/src/regression/core.test.ts:79]: Could you explain this change? Using `innerText` is generally an antipattern

Comment (uwedeportivo) [reply web/src/regression/core.test.ts:79]: with .textContent i get the text and then some crap behind it that looks like css style names:

```
"{ /* These are new settings */}.monaco-list.list_id_2:focus .monaco-list-row.focused { background-color: #d6ebff; }\n.monaco-list.list_id_2:focus .monaco-list-row.focused:hover { background-color: #d6ebff; }\n.monaco-list.list_id_2:focus .monaco-list-row.selected { background-color: #0069d1; }\n.monaco-list.list_id_2:focus .monaco-list-row.selected:hover { background-color: #0069d1; }\n.monaco-list.list_id_2:focus .monaco-list-row.selected { color: #ffffff; }\n\n\t\t\t\t.monaco-drag-image,\n\t\t\t\t.monaco-list.list_id_2:focus .monaco-list-row.selected.focused { background-color: #0074e8; }\n\t\t\t\n\n\t\t\t\t.monaco-drag-image,\n\t\t\t\t.monaco-list.list_id_2:focus .monaco-list-row.selected.focused { color: #ffffff; }\n\t\t\t\n.monaco-list.list_id_2 .monaco-list-row.focused { background-color:  #f2f4f8; }\n.monaco-list.list_id_2 .monaco-list-row.focused:hover { background-color:  #f2f4f8; }\n.monaco-list.list_id_2 .monaco-list-row.selected { background-color:  #e4e6f1; }\n.monaco-list.list_id_2 .monaco-list-row.selected:hover { background-color:  #e4e6f1; }\n.monaco-list.list_id_2:not(.drop-target) .monaco-list-row:hover:not(.selected):not(.focused) { background-color:  #f2f4f8; }\n\n\t\t\t\t.monaco-list.list_id_2.drop-target,\n\t\t\t\t.monaco-list.list_id_2 .monaco-list-row.drop-target { background-color: #d6ebff !important; color: inherit !important; }\n\t\t\t\n.monaco-list-type-filter { background-color: #efc1ad }\n.monaco-list-type-filter { border: 1px solid rgba(0, 0, 0, 0); }\n.monaco-list-type-filter.no-matches { border: 1px solid #be1100; }\n.monaco-list-type-filter { box-shadow: 1px 1px 1px #a8a8a8; }
```

Comment (uwedeportivo) [reply web/src/regression/core.test.ts:79]: ¯\_(ツ)_/¯

Comment (felixfbecker) [reply web/src/regression/core.test.ts:79]: Sounds like you have the wrong element then, one that also includes a `<style>` element

File: web/src/regression/core.test.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ describe('Core functionality regression test suite', () => {
             await driver.page.waitForSelector('.e2e-settings-file .monaco-editor')
             return driver.page.evaluate(() => {
                 const editor = document.querySelector('.e2e-settings-file .monaco-editor') as HTMLElement
```
Comment (felixfbecker) [inline web/src/regression/core.test.ts:78]: This element should work and not include the `<style>`:
```suggestion
                const editor = document.querySelector<HTMLElement>('.e2e-settings-file .monaco-editor .view-lines')
```

Comment (uwedeportivo) [reply web/src/regression/core.test.ts:78]: cool, thanks felix. will use :-)

**Comment by uwedeportivo** [comment]
@felixfbecker since i have your attention: do you have some idea why puppeteer fails to find the "Codecov 80%" element 

https://github.com/sourcegraph/sourcegraph/blob/master/web/src/regression/extensions.test.ts#L119

i can confirm it's there and working

**Comment by uwedeportivo** [comment]
ok, will try later. i'm tied up now with other stuff related to a customer and to the release. thanks for looking

**Comment by uwedeportivo** [comment]
image

@felixfbecker your fix for innerText/ytextContent worked. thanks for that

**Comment by felixfbecker** [comment]
Can this be closed?

**Pull request closed without merging**